packages/dev/core/src/Materials/Textures/videoTexture.ts

```diff
@@ -267,15 +268,13 @@ export class VideoTexture extends Texture {
 
         return video;
     }
+    private _videoResizeEventListener = (): void => {
+        this._resizeInternalTexture(false /* skipDispose */);
+    }
 
-    private _createInternalTexture = (): void => {
-        if (this._texture != null) {
-            if (this._displayingPosterTexture) {
-                this._texture.dispose();
-                this._displayingPosterTexture = false;
-            } else {
-                return;
-            }
+    private _resizeInternalTexture = (skipDispose: boolean): void => {
+        if (this._texture != null && !skipDispose) {

```

Comment:
This might leak ??? if skipDispose is true _texture is replaced but not disposed ?

Reply:
Good point. I've updated the latest iteration so that `_resizeInternalTexture` now always disposes `_texture` if it is non-null.